internal/starter/loadConfig.go

```diff
@@ -0,0 +1,220 @@
+package starter
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"reflect"
+	"strconv"
+	"time"
+)
+
+var (
+	ErrStructValPrt = errors.New("expected struct pointer")
+)
+
+type EnvTypes interface {
+	~int | ~int8 | ~int16 | ~int32 | ~int64 | ~uint | ~uint8 | ~uint16 | ~uint32 | ~uint64 | ~float32 | ~float64 | ~string
+}
+
+func ConvertStringToType[T EnvTypes](valueStr string) T {
+	var value T
+	switch p := any(&value).(type) {
+	case *int:
+		i, _ := strconv.ParseInt(valueStr, 10, 0)
+		*p = int(i)
+		return value
+	case *int8:
+		i, _ := strconv.ParseInt(valueStr, 10, 8)
+		*p = int8(i)
+		return value
+	case *int16:
+		i, _ := strconv.ParseInt(valueStr, 10, 16)
+		*p = int16(i)
+		return value
+	case *int32:
+		i, _ := strconv.ParseInt(valueStr, 10, 32)
+		*p = int32(i)
+		return value
+	case *int64:
+		i, _ := strconv.ParseInt(valueStr, 10, 64)
+		*p = i
+		return value
+	case *uint:
+		i, _ := strconv.ParseUint(valueStr, 10, 0)
+		*p = uint(i)
+		return value
+	case *uint8:
+		i, _ := strconv.ParseUint(valueStr, 10, 8)
+		*p = uint8(i)
+		return value
+	case *uint16:
+		i, _ := strconv.ParseUint(valueStr, 10, 16)
+		*p = uint16(i)
+		return value
+	case *uint32:
+		i, _ := strconv.ParseUint(valueStr, 10, 32)
+		*p = uint32(i)
+		return value
+	case *uint64:
+		i, _ := strconv.ParseUint(valueStr, 10, 64)
+		*p = i
+		return value
+	case *float32:
+		i, _ := strconv.ParseFloat(valueStr, 32)
+		*p = float32(i)
+		return value
+	case *float64:
+		i, _ := strconv.ParseFloat(valueStr, 64)
+		*p = i
+		return value
+	case *string:
+		*p = valueStr
+		return value
+	default:
+		return any(valueStr).(T)
+	}
+}
+
+func GetEnv[T EnvTypes](key string, fallback T) T {
+	envValue := os.Getenv(key)
+	if envValue == "" {
+		return fallback
+	}
+	var value T
+	switch p := any(&value).(type) {
```

Comment:
This looks like something that could potentially exist as some library function somewhere (either in the stdlib or some package).

It would be good to do a research.

Reply:
https://github.com/caarlos0/env is used to load environment to config struct